Sources/Noora/Components/CollapsibleStep.swift

```diff
@@ -0,0 +1,130 @@
+import Foundation
+import Rainbow
+
+struct CollapsibleStep {
+    // MARK: - Attributes
+
+    let title: TerminalText
+    let successMessage: TerminalText?
+    let visibleLines: UInt
+    let errorMessage: TerminalText?
+    let task: (@escaping (TerminalText) -> Void) async throws -> Void
+    let theme: Theme
+    let terminal: Terminaling
+    let renderer: Rendering
+    let standardPipelines: StandardPipelines
+
+    init(
+        title: TerminalText,
+        successMessage: TerminalText?,
+        errorMessage: TerminalText?,
+        visibleLines: UInt,
+        task: @escaping (@escaping (TerminalText) -> Void) async throws -> Void,
+        theme: Theme,
+        terminal: Terminaling,
+        renderer: Rendering,
+        standardPipelines: StandardPipelines
+    ) {
+        self.title = title
+        self.successMessage = successMessage
+        self.errorMessage = errorMessage
+        self.visibleLines = visibleLines
+        self.task = task
+        self.theme = theme
+        self.terminal = terminal
+        self.renderer = renderer
+        self.standardPipelines = standardPipelines
+    }
+
+    func run() async throws {
+        if terminal.isInteractive {
+            try await runInteractive()
+        } else {
+            try await runNonInteractive()
+        }
+    }
+
+    func runNonInteractive() async throws {
+        standardPipelines.output
+            .write(
+                content: "◉ \(title.formatted(theme: theme, terminal: terminal)) \n"
```

Comment:
```suggestion
                content: "◉ \(title.formatted(theme: theme, terminal: terminal))\n"
```

nit: I'd remove the extra trailing space.